src/gui/sdlmain.cpp

```diff
@@ -3386,7 +3387,132 @@ static void set_output(Section *sec, bool should_stretch_pixels)
+{
+	// Variables common to all screen-modes
+	const auto renderer = SDL_GetRenderer(sdl.window);
+	const auto canvas   = get_canvas_size(sdl.desktop.type);
+
+#if C_OPENGL
+	// Get the OpenGL-renderer surface
+	// -------------------------------
+	if (sdl.desktop.type == SCREEN_OPENGL) {
+		// Setup our OpenGL image properties
+		constexpr int gl_channels       = 3; // RBG (no alpha)
+		constexpr int gl_bits_per_pixel = gl_channels * 8; // 8-bpp
+		const size_t bytes_per_row      = gl_channels * canvas.w;
+
+		// Allocate a 24-bit surface to be populated
+		const auto surface = SDL_CreateRGBSurface(SDL_SWSURFACE,
+		                                          canvas.w,
+		                                          canvas.h,
+		                                          gl_bits_per_pixel,
+		                                          RMASK,
+		                                          GMASK,
+		                                          BMASK,
+		                                          AMASK);
+		if (!surface) {
+			LOG_WARNING("SDL: Failed creating a surface for OpenGL because %s",
+			            SDL_GetError());
+			return {};
+		}
+		// Per OpenGL's documentation:
+		// The glReadPixels function starts at the lower left corner: (x
+		// + i, y + j) and iterates for 0 <= i < width and 0 <= j <
+		// height. It describes the pxiels as being "the i'th pixel in
+		// the j'th row". Pixels are returned in row-order from the
+		// lowest to the highest row, left to right in each row.
+		std::vector<uint8_t> pixels(bytes_per_row * canvas.h);
+		glReadPixels(0, 0, canvas.w, canvas.h, GL_RGB, GL_UNSIGNED_BYTE, pixels.data());
+
+		// To match SDL's surface ordering, we invert the rows (outer)
+		// and lines (inner):
+		auto surface_pixels = static_cast<char *>(surface->pixels);
+		for (int j = 0; j < canvas.h; ++j) {
+			auto target_row = surface_pixels + surface->pitch * j;
+			const auto source_row = pixels.data() +
+			                        bytes_per_row * (canvas.h - j - 1);
+			memcpy(target_row, source_row, bytes_per_row);
+		}
+		return surface;
+	}
+#endif
+
+	// Get the SDL texture-renderer surface
+	// ------------------------------------
+	else if (sdl.desktop.type == SCREEN_TEXTURE) {
+		// Get the renderer's pixel format
+		SDL_RendererInfo rinfo;
+		if (SDL_GetRendererInfo(renderer, &rinfo) != 0) {
+			LOG_MSG("SDL: Failed to get a rendering info because %s",
+			        SDL_GetError());
+			return {};
+		}
+		const auto pixel_format = rinfo.texture_formats[0];
+
+		// Create a 32-bit surface with color format matching the renderer
+		const auto surface = SDL_CreateRGBSurfaceWithFormat(
+		        SDL_SWSURFACE, canvas.w, canvas.h, 32, pixel_format);
+		if (!surface || !renderer) {
+			LOG_WARNING("SDL: Failed creating a surface because %s",
+			            SDL_GetError());
+			return {};
+		}
+		// Copy the pixels from the renderer into our new surface
+		if (SDL_RenderReadPixels(renderer,
```

Comment:
(resolving to go ahead w/ the merge - not because this comment is resolved.  I've added a comment to the PR itself below referencing this ticket and asking for help)